provider/aws: Crash plan aws_sns_topic resource with empty policy

### Terraform Version
Terraform v0.7.3

### Affected Resource(s)
- aws_sns_topic

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "ap-northeast-1"
}

resource "aws_sns_topic" "default" {
  name            = "foobarbaz"
  policy          = ""
}
```

### Panic Output

```
aws_sns_topic.default: Refreshing state... (ID: arn:aws:sns:ap-northeast-1:328187951919:foobarbaz)

Error running plan: 1 error(s) occurred:

panic: runtime error: invalid memory address or nil pointer dereference
2016/09/13 20:36:27 [DEBUG] plugin: terraform: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xf80242]
2016/09/13 20:36:27 [DEBUG] plugin: terraform: 
2016/09/13 20:36:27 [DEBUG] plugin: terraform: goroutine 114 [running]:
2016/09/13 20:36:27 [DEBUG] plugin: terraform: panic(0x284f000, 0xc420014090)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/go/src/runtime/panic.go:500 +0x1a1
2016/09/13 20:36:27 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.schemaMap.diff(0xc4203ae2d0, 0x2e07e5a, 0x6, 0xc4207248c0, 0xc4205a26a0, 0xc4204fa000, 0x0, 0x0, 0x0)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/schema.go:641 +0x1d2
2016/09/13 20:36:27 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.schemaMap.Diff(0xc4203ae2d0, 0xc420596000, 0xc4205b05d0, 0xc4205a4530, 0xc42006a100, 0x0)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/schema.go:348 +0x1d6
2016/09/13 20:36:27 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Diff(0xc42070ec60, 0xc420596000, 0xc4205b05d0, 0xd, 0xc42006a1c0, 0xc4205b0601)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:179 +0x42
2016/09/13 20:36:27 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Diff(0xc4203ca1b0, 0xc42056a300, 0xc420596000, 0xc4205b05d0, 0x0, 0x18, 0x18)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:191 +0x84
2016/09/13 20:36:27 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Diff(0xc420564480, 0xc4205a2020, 0xc4205a4520, 0x0, 0x0)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:471 +0x57
2016/09/13 20:36:27 [DEBUG] plugin: terraform: reflect.Value.call(0xc4202ba7e0, 0xc4202c0768, 0x13, 0x2e04768, 0x4, 0xc420321ee0, 0x3, 0x3, 0x0, 0x0, ...)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:434 +0x5c8
2016/09/13 20:36:27 [DEBUG] plugin: terraform: reflect.Value.Call(0xc4202ba7e0, 0xc4202c0768, 0x13, 0xc420321ee0, 0x3, 0x3, 0x0, 0x0, 0x0)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/go/src/reflect/value.go:302 +0xa4
2016/09/13 20:36:27 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc4202b3ec0, 0xc4202b3e80, 0xc420572c98, 0xc42063d380, 0xc42054d760, 0x25b56e0, 0xc4205a2020, 0x16, 0x25b5720, 0xc4205a4520, ...)
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:383 +0x148
2016/09/13 20:36:27 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/09/13 20:36:27 [DEBUG] plugin: terraform:  /opt/go/src/net/rpc/server.go:477 +0x421
2016/09/13 20:36:27 [ERROR] root: eval: *terraform.EvalDiff, err: unexpected EOF
2016/09/13 20:36:27 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2016/09/13 20:36:27 [ERROR] root: eval: *terraform.EvalOpFilter, err: unexpected EOF
2016/09/13 20:36:27 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2016/09/13 20:36:27 [TRACE] [walkPlan] Exiting eval tree: aws_sns_topic.default
2016/09/13 20:36:27 [DEBUG] vertex provider.aws (close), got dep: aws_sns_topic.default
2016/09/13 20:36:27 [DEBUG] plugin: waiting for all plugin processes to complete...
2016/09/13 20:36:27 [DEBUG] plugin: /home/kkojima/bin/terraform: plugin process exited
```
[crash.log](https://gist.github.com/kjmkznr/c6206f739dd747ae641a69eb1c5e8755)

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`

### References
- GH-8650
- GH-8585

--- a/helper/schema/schema.go
+++ b/helper/schema/schema.go
@@ -638,7 +638,9 @@ func (m schemaMap) diff(
 	}
 
 	for attrK, attrV := range unsupressedDiff.Attributes {
-		if schema.DiffSuppressFunc != nil && schema.DiffSuppressFunc(attrK, attrV.Old, attrV.New, d) {
+		if schema.DiffSuppressFunc != nil &&
+			attrV != nil &&
+			schema.DiffSuppressFunc(attrK, attrV.Old, attrV.New, d) {
 			continue
 		}
 

